packages/core/src/components/steps-modal/steps-modal.tsx

```diff
@@ -0,0 +1,100 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  Host,
+  Prop,
+  forceUpdate,
+  h,
+} from '@stencil/core'
+
+@Component({
+  tag: 'atom-steps-modal',
+  styleUrl: 'steps-modal.scss',
+  shadow: false,
+})
+export class AtomStepsModal {
+  @Prop() steps: number = 0
+  @Prop({ mutable: true }) currentStep: number = 0
+  @Prop() trigger?: string
+  @Prop() stepsTitles: string
+  @Prop({ mutable: true }) isOpen: boolean = false
+
+  @Event() atomFinish: EventEmitter
+  @Event() atomCancel: EventEmitter
+  @Event() atomNextStep: EventEmitter
+  @Event() atomPreviousStep: EventEmitter
+  @Event() atomCloseClick: EventEmitter
+  @Event() atomDidDismiss: EventEmitter
+  @Event() atomDidPresent: EventEmitter
+
+  @Element() el!: HTMLElement
+
+  private stepsTitlesArray: string[] = []
+
+  componentWillLoad() {
+    this.stepsTitlesArray = this.stepsTitles.split(',')
+  }
+
+  private handleStep = (step: number) => {
+    this.currentStep = step
+  }
+
+  private handlePrimaryClick = () => {
+    if (this.currentStep === this.steps - 1) {
+      this.atomFinish.emit()
+    } else {
+      this.handleStep(this.currentStep + 1)
+      this.atomNextStep.emit()
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**question:** ‏maybe we could emit "next" and "previous" step events with actual step
```js
this.atomNextStep.emit(this.currentStep + 1)
```